src/core.rs

```diff
@@ -297,7 +297,10 @@ impl<T> Webauthn<T> {
         }
 
         // Verify that the value of C.origin matches the Relying Party's origin.
-        if data.client_data_json.origin != self.config.get_origin() {
+        if !self.config.allow_cross_origin()
```

Comment:
I think this is not correct. https://www.w3.org/TR/webauthn-2/#sctn-verifying-assertion does not actually inspect crossOrigin during the assertion or registration, and we still need to check the origin is our rp origin.

So this makes me question, what is even the point of this boolean if we never inspect it during a registration/assertion? 

@agrinman Thoughts?

Reply:
I think there's a bit [here](https://www.w3.org/TR/webauthn-2/#clientdatajson-verification) about limited verification if you cannot support a full JSON parser?

Reply:
> Verifiers may use the following algorithm to verify an encoded CollectedClientData if they cannot support a full JSON parser

Reply:
Step 4 there still states:

```
A string, origin, that contains the expected origin that issued the request to the user agent.
```

The issue is that your logic as it's set, if you allow cross_origin, then the origin value isn't checked. This is incorrect.

The code should be:

```
if data.client_data_json.origin != self.config.get_origin() {
    // as before
}

if !self.config.allow_cross_origin() && data.client_data_json.cross_origin.unwrap_or(false) {
    // return error, dis allowing cross origin ccd
}
```

Reply:
That makes sense